### DISCUSSION TITLE
Cypress XHR 400 Error when clicking a button. Running cypress securely.

### DISCUSSION THREAD

**Discussion by PwntasticKev** [cypress-io/cypress#15008] [Questions and Help]
Hi! I've been working with cypress for one day now. I have a button that will refresh the page once you click on it and some tests after that still go through. 

clicking on this button will throw a 400 error. ( post request )

I'm able to click other buttons on the page and can also manually do it within the browser in a new tab. but once the test automates it, it will fail.

I read a few cases:
[#1951](https://github.com/cypress-io/cypress/issues/1951)
[#2001](https://github.com/cypress-io/cypress/issues/2001)

but none seem to be able to solve this issue.

*index.js*
```
module.exports = (on, config) => {
  on('before:browser:launch', (browser, launchOptions) => {
    if (browser.name === 'chrome') {
      launchOptions.args.push('--disable-site-isolation-trials');
      launchOptions.args.push('--incognito')
      return launchOptions
    }
  })
}
```

*test.js*

```
context('Checkout Purchase', () => {
	describe('Checkout Form', () => {
		it('Tests Promo Codes', () => {
			cy.wait(2000);
			cy.get('#promoCode')
			.type('bw-nw-testing')
			.should('have.value', 'bw-nw-testing')
			cy.get('#applyPromoCodeButton')
			  .click()
			  .wait(2000)
		})
        })
})
```

*cypress.json*

```
{
  "projectId": "lkj8fdj",
  "chromeWebSecurity": false,
  "baseUrl": "http://localhost:3000"
}
```

**Comment by cbruge** [comment]
Hey,

if it works manually and not with Cypress you should open the network tab and compare the request sent when you do it manually vs when you do it in Cypress. 
Cypress also starts with a clean cache and you might need to whitelist some cookies, see https://docs.cypress.io/api/cypress-api/cookies.html#Syntax
